docs/index.md

```diff
@@ -0,0 +1,187 @@
+- the behaviour is relying entirely on business facts 
+
+If the answer is yes for both of them, then creating a ruleset is a good idea. 
+
+
+In the current release of Regula (`0.5.0`) stakeholders can only create or edits rules through scripts which require some engineering knowledge. Later version will include a proper UI for stakeholders to interact with Regula in a simple manner. 
+
+
+## Components overview 
+
+> ⚠ TODO: we should discuss what we want to bring in here, this can get quite complex really quick.
+
+![Cinematic](./cinematic.png)
+
+- Regula clients are querying Regula server
+  - they live in the source code of a given application
+  - when high performances are required, i.e. ruleset evaluations should be as fast as possible, there are caching mechanism available in some clients. See more about this in the technical documentation.
+  
+- Regula server answers requests from the clients. 
+- Rulesets storage is where the rulesets and their versions are being kept. It is not supposed to be interacted with directly.
+
+## Design principles
+
+Rules engines are delicate beasts. It is important to remember that technically, it's about extracting behaviour from the source code and relocating it outside in a more "accessible" place that is outside of any continuous integration mechanism. 
+
+Regula takes an opinionated approach to lower the risks and make it harder to break from expected behaviours. Regula's team objective is to focus on a curated feature set, putting reliability and predictability at a top priority. 
+
+## Predictability 
+
+Regula, by splitting the ownership of a given functionality of the software that uses it, needs to be predictable. Behaviour being owned by whoever edits the rules, it is mandatory that a given ruleset is not able to break the software that uses it. 
+
+On developers side:
+
+Regula is "typed" and that is why the concept of **signature** exists, illustrated by:
+
+```
+float64 /marketplace/radius(string city, string driver-status)
+```
+
+By doing so, it allows developers of both typed and dynamic languages to be certain that they always get a value of a given type. Changing the signature implies to create a new rule and such a break will logically require to edit the source code of the application using Regula to reflect the change. 
+
+On stakeholders side: 
+
+Regula rulesets are "versioned", meaning every updates to rulesets are tracked. By doing so, it is possible for stakeholders to observe how the behaviour is evolving. 
+
+Moreover, versioning allows stakeholders to to take versions in consideration while designing a product. A given business object could be stored with the version of rulesets it is intended to use throughout its life, making it. If a given promotion coupon for example had been applied on a product being sold, by storing the version used, it can easily be computed again against that version, allowing to get the same exact result. 
```

Comment:
"to to" should be "to".  "in consideration" should be "into consideration", and "while designing" should be "when designing".    The ", making it" at the end of the 2nd sentence should be removed.   In the 3rd sentence "For example, " should be at the start of the sentence.   "applied on" should be "applied to".